src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -0,0 +1,1706 @@
+                events.ScheduleEvent(EVENT_DESPAWN, 1000);
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            events.Update(diff);
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_THROW_ROCK:
+                        if (!me->IsWithinMeleeRange(me->GetVictim()))
+                            DoCastVictim(SPELL_THROW_ROCK);
+                        events.ScheduleEvent(EVENT_THROW_ROCK, 2500);
+                        break;
+                    case EVENT_SWITCH_POLE:
+                        if (!me->IsInCombat())
+                        {
+                            // This stores objects that are too far away due to big combat reach
+                            me->GetCreatureListWithEntryInGrid(PolesList, NPC_BALANCE_POLE_1, 1.0f);
+                            me->GetCreatureListWithEntryInGrid(PolesList2, NPC_BALANCE_POLE_2, 1.0f);
+                            // Join both lists with possible different NPC entries
+                            PolesList.splice(PolesList.end(), PolesList2);
+                            // Convert list to vector, so we can access iterator to be able to shuffle the list
+                            std::vector<Creature*> BalancePolesList {std::make_move_iterator(std::begin(PolesList)), std::make_move_iterator(std::end(PolesList))};
+                            // Shuffle the list so NPCs won't jump always on the same poles
+                            Trinity::Containers::RandomShuffle(BalancePolesList);
+
+                            for (std::vector<Creature*>::const_iterator itr = BalancePolesList.begin(); itr != BalancePolesList.end(); ++itr)
+                            {
+                                Position offset;
+                                offset.m_positionX = fabsf((*itr)->GetPositionX() - me->GetPositionX());
+                                offset.m_positionY = fabsf((*itr)->GetPositionY() - me->GetPositionY());
+
+                                // Object is too far
+                                if (offset.m_positionX > 5.0f || offset.m_positionY > 5.0f)
+                                    continue;
+
+                                if (!(*itr)->HasAura(SPELL_MONK_RIDE_POLE) && !(*itr)->HasAura(SPELL_RIDE_VEHICLE_POLE))
+                                {
+                                    (*itr)->CastSpell(me, SPELL_FORCECAST_RIDE_POLE, true);
+                                    break;
+                                }
+                            }
+                            events.ScheduleEvent(EVENT_SWITCH_POLE, urand(15000, 30000));
+                        }
+                        break;
+                    case EVENT_DESPAWN:
+                        // Transform is casted only when in frog pool
+                        if (Creature* creature = me->FindNearestCreature(NPC_CURSED_POOL_CONTROLLER, 71.0f, true))
```

Comment:
```
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Pandaria/zone_the_wandering_isle.cpp:624:39:
 fatal error: unused variable 'creature' [-Wunused-variable]
    if (Creature* creature = me->FindNearestCreature(NPC_CURSED_POOL_CONTROLLER, 71.0f, true))
                  ^
```
Is it possible to use `if me->FindNearestCreature(NPC_CURSED_POOL_CONTROLLER, 71.0f, true)` instead?

Reply:
yes, corrected